### PR TITLE
User defined vector definition in SimulationTimeSeries plugin

### PR DESCRIPTION
**Pull request by jorgenherje**
Add user defined vector definitions for `SimulationTimeSeries`- plugin

---
- Add configurable user defined vector definitions for plugin 
- Rename `AVG_` and `INVTL_` to `PER_DAY_` and `PER_INTVL_`, respectively. For vector it is calculated from
    - **NOTE**: Does not replace `T`-suffix with `R`-suffix for `PER_DAY_`-vector
    - Previously `WOPT` gave `AVG_WOPR` and `INTVL_WOPT`. Not it gives `PER_DAY_WOPT` and `PER_INTVL_WOPT`
- Remove `webviz_subsurface/_abbreviations/abbreviation_data/reservoir_simulation_vectors.json` and retrieve `VectorDefinitions` using new in Python-API for `webviz-subsurface-components`.
---

## NOTE:
Resolves:
* https://github.com/equinor/webviz-subsurface/issues/321

Dependent on: 
* https://github.com/equinor/webviz-subsurface-components/pull/782
* https://github.com/equinor/webviz-subsurface-testdata/pull/126

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -15,13 +15,15 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
 
-- [#935](https://github.com/equinor/webviz-subsurface/pull/935) - Deprecated plugin `ReservoirSimulationTimeSeries`. This has been replaced by the faster, more flexible and less memory hungry plugin `SimulationTimeSeries`.
 
 ### Changed
 
+- [#940](https://github.com/equinor/webviz-subsurface/pull/940) - `SimulationTimeSeries` - Changed vector annotation from "AVG_" with suffix "R" and "INTVL_" to "PER_DAY_" and "PER_INTVL_". Retrieve `VectorDefinitions` via Python-API for `webviz-subsurface-components`.
 - [#924](https://github.com/equinor/webviz-subsurface/pull/924) - Improvements to the `ParameterFilter` functionality, e.g information regarding active filters and which relizatons are filtered out, and better handling of multiple ensembles. Improvements to the `ParameterAnalysis` plugin, e.g. added boxplot, fixed table formatting and update of parameters based on selected ensemble.
+- [#935](https://github.com/equinor/webviz-subsurface/pull/935) - Deprecated plugin `ReservoirSimulationTimeSeries`. This has been replaced by the faster, more flexible and less memory hungry plugin `SimulationTimeSeries`.
```
Comment (anders-kiaer) [inline CHANGELOG.md:26]: We need to move this now to `UNRELEASED` section.

File: webviz_subsurface/_utils/vector_selector.py [hunk 2]
```diff
@@ -15,16 +18,17 @@ def add_vector_to_vector_selector_data(
...
-                    "children": [] if index < len(nodes) - 1 else None,
-                }
-            )
+            node_data: dict = {
+                "name": node,
+                "children": [] if index < len(nodes) - 1 else None,
+            }
+            if not description_at_last_node and description and index == 0:
+                node_data["description"] = description
+            if description_at_last_node and description and (index == len(nodes) - 1):
```
Comment (anders-kiaer) [inline webviz_subsurface/_utils/vector_selector.py:27]:
```suggestion
            elif description_at_last_node and description and (index == len(nodes) - 1):
```

File: webviz_subsurface/plugins/_simulation_time_series/_plugin.py [hunk 3]
```diff
@@ -62,6 +72,25 @@ def __init__(
         self._webviz_settings = webviz_settings
         self._obsfile = obsfile
 
+        # Retreive user defined vector descriptions from configuration and validate
```
Comment (anders-kiaer) [inline webviz_subsurface/plugins/_simulation_time_series/_plugin.py:75]:
```suggestion
        # Retrieve user defined vector descriptions from configuration and validate
```

**Review by anders-kiaer — CHANGES_REQUESTED** [review]
Good work! Only a couple of minor comments from my side. I like the new tests!

File: CHANGELOG.md (hunk 1 above)
Comment (jorgenherje) [reply CHANGELOG.md:26]: Aren't these already placed under "Added" and "Changed" category in `UNRELEASED`-section?

Comment (jorgenherje) [reply CHANGELOG.md:31]: My bad, got the point after merging master into branch!

**Review by anders-kiaer — APPROVED** [review]

**Pull request merged**